.github/workflows/PR-build.yml

```diff
@@ -0,0 +1,144 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: PR Build
+on:
+  pull_request:
+    branches: 
+      - main 
+    types: 
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths-ignore:
+      - '.github/**'
+      - '!.github/workflows/CI.yml'
+      - '!.github/workflows/CI-Operator.yml'
+      - '**.md'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  macos-unittest:
+    runs-on: macos-latest
+    steps:
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v3
+      with:
+        go-version: ~1.19.2
+
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Cache build output
+      id: cached_binaries
+      uses: actions/cache@v3
+      with:
+        path: |
+              ~/Library/Caches/go-build
+              ~/go/pkg/mod
+        key: v1-go-pkg-mod-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+
+    - name: Test
+      if: steps.cached_binaries.outputs.cache-hit != 'true'
+      run: make test
+
+    - name: Upload coverage to Codecov
+      if: steps.cached_binaries.outputs.cache-hit != 'true'
+      uses: codecov/codecov-action@v3
+      with:
+        verbose: true
+
+    - name: Build
+      if: steps.cached_binaries.outputs.cache-hit != 'true'
+      run: make build
+  
+  windows-unittest:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```

Comment:
Are we failing fast or not? The description says fail-fast is true. This says false, and is the only place where `fail-fast` is set.

Reply:
NIce catch. I have edited the fail fast to false. However, it still upholds the statement:
```
 If one fail, the others won't affect each other and continue to run
```
Yes since its the only [place we use matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#handling-failures). However, I can also include others jobs (e.g linux and mac-os) as a part of Windows matrix to share the same benefit. The only reason for separating is for clear separation only .